### ISSUE TITLE
How to run the models Fully locally.(in layman's term No WIFI)

### ISSUE THREAD

**Issue by cm5kZGV2MjAyM3B1YmdpdGh1YmFjYw** [huggingface/huggingface_hub#1508]
### Describe the bug

from_pretrained() by default just stupidly pings server not stop and throws exception when fail due to code without proper error handling.
eg>requests.exceptions.ConnectionError: HTTPSConnectionPool(host='huggingface.co', port=443): Max retries exceeded with url: /api/models

even if i do from_pretrained(cache_dir="someplace")
and itside it had already donwloaded the model.It will still pings the server.

How do I just make it run using my local model and dont ever try to ping server,unless I enable it,its wasting my time.

If such feature havent come across thousands of users and developers then....
Can I make a propose to just make an api that is by default prioritize local environment over online and HANDLE EXCEPTION PROPERLY with a proper feedback with a valid solution.

### Reproduction

just try out any model that is working ,
run it once and lets it download into your device.(optional)
disconnect from internet.
rerun

### Logs

```shell
lib\site-packages\huggingface_hub\hf_api.py", line 1675, in model_info
    r = get_session().get(path, headers=headers, timeout=timeout, params=params)
....
site-packages\requests\adapters.py", line 519, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='huggingface.co', port=443): Max retries exceeded with url: /api/models/runwayml/stable-diffusion-v1-5 (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x000001FAEDD00280>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))
```

### System info

```shell
- cpy 3.10
```

**Comment by Wauplin** [comment]
Hey :wave: If you want to run fully offline, you can set `HF_HUB_OFFLINE=1` as environment variable (see [docs](https://huggingface.co/docs/huggingface_hub/package_reference/environment_variables#hfhuboffline)). But to be honest, this mostly depend on which library you're using. Both `transformers` and `diffusers` respects this env variable but in your case I don't know what lib you're using. `huggingface_hub` is just the underlying lib making calls to the Hub. Its purpose is not to know how to load local files (which is library-dependent).

Also, please remember to be considerate and respectful when interacting with open-source's maintainers. Calling stuff _stupid_ and using capital letters is never a good approach when trying to seek help.

**Comment by cm5kZGV2MjAyM3B1YmdpdGh1YmFjYw** [comment]
noted with thanks,it worked.
